plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/functions/tablechanges/TableChangesFunction.java

```diff
@@ -115,6 +116,9 @@ public TableFunctionAnalysis analyze(ConnectorSession session, ConnectorTransact
 
         checkSnapshotExists(icebergTable, startSnapshotId);
         checkSnapshotExists(icebergTable, endSnapshotId);
+        if (!SnapshotUtil.isParentAncestorOf(icebergTable, endSnapshotId, startSnapshotId)) {
```

Comment:
Change the commit message.
We're fixing an existing functionality here, not only adding a test.